src/Symfony/Component/Messenger/Handler/ExactlyOneHandlerLocator.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Handler;
+
+use Symfony\Component\Messenger\Envelope;
+use Symfony\Component\Messenger\Exception\MultipleHandlersForMessageException;
+use Symfony\Component\Messenger\Exception\NoHandlerForMessageException;
+
+final class ExactlyOneHandlerLocator implements HandlersLocatorInterface
+{
+    private $handlers;
+
+    public function __construct(array $handlers)
+    {
+        foreach ($handlers as $type => $h) {
+            if (\count($h) > 1) {
+                throw new MultipleHandlersForMessageException($type);
+            }
+        }
+
+        $this->handlers = $handlers;
+    }
+
+    public function getHandlers(Envelope $envelope): iterable
+    {
+        $type = \get_class($envelope->getMessage());
+        if (!isset($this->handlers[$type])) {
+            throw new NoHandlerForMessageException($type);
```

Comment:
Shouldn't the error message include the message type, rather that the type/class name
```
throw new NoHandlerForMessageException($this->handlers[$type]);
```
Theory being the error message 'Multiple handlers for message "%s"',  thus, does this mean $envelope->getMessage()?

Reply:
In this specific case, `$type` holds the message type and `$this->handlers[$type]` is not set.

> Theory being the error message 'Multiple handlers for message "%s"', thus, does this mean $envelope->getMessage()?

I don't understand what you mean.